policy/pa.go

```diff
@@ -111,6 +113,50 @@ func (pa *AuthorityImpl) loadHostnamePolicy(b []byte) error {
 	return nil
 }
 
+// SetChallengesWhitelistFile will load the given whitelist file, returning error if it
+// fails. It will also start a reloader in case the file changes.
+func (pa *AuthorityImpl) SetChallengesWhitelistFile(f string) error {
+	_, err := reloader.New(f, pa.loadChallengesWhitelist, pa.challengesWhitelistLoadError)
+	return err
+}
+
+func (pa *AuthorityImpl) challengesWhitelistLoadError(err error) {
+	pa.log.AuditErr(fmt.Sprintf("error loading challenges whitelist: %s", err))
+}
+
+func (pa *AuthorityImpl) loadChallengesWhitelist(b []byte) error {
+	hash := sha256.Sum256(b)
+	pa.log.Info(fmt.Sprintf("loading challenges whitelist, sha256: %s",
+		hex.EncodeToString(hash[:])))
+	var wl map[string]interface{}
+	err := json.Unmarshal(b, &wl)
+	if err != nil {
+		return err
+	}
+
+	chalWl := make(map[string]map[int64]bool)
+
+	for k, v := range wl {
+		chalWl[k] = make(map[int64]bool)
+		list, ok := v.([]string)
+		if !ok {
+			return fmt.Errorf("Malformed whitelist entry: %v", v)
+		}
+		for _, s := range list {
+			i, err := strconv.ParseInt(s, 10, 0)
```

Comment:
There's no need to use parseInt here, since we can just represent the integers directly as integers in the JSON (Go handles integers > 53 bits just fine even thought JS doesn't).